src/messages/MOSDOp.h

```diff
@@ -242,14 +236,14 @@ struct ceph_osd_request_head {
       ::encode_nohead(snaps, payload);
     } else {
       header.version = HEAD_VERSION;
-      ::encode(client_inc, payload);
+      ::encode(pgid, payload);
```

Comment:
We need to be backwards-compatible in userspace as well. If somebody upgrades their client then it will encode with this reordered version, but if they have an old OSD it won't work properly.

We can deal with that, and do so not infrequently: we need to
*) set the minimum decode version to our new (bumped) encode version when encoding with the reordering
*) create a new feature bit for "supports partial decode" or whatever
*) check that feature bit when encoding an MOSDOp message for somebody and conditionally encode our new reordered version or the previous version based on which one they support.
*) When decoding, check ahead of time if it's a version with the reordered members or an older version, and branch as appropriate.

You can look at the OSDMap::encode and decode functions for some of the more extensive examples of how we handle these issues.